app/models/field/complex_datation.rb

```diff
@@ -143,13 +143,23 @@ def edit_props(item)
         active: choice_set.not_deactivated? && choice_set.not_deleted?,
         newChoiceModalUrl: Rails.application.routes.url_helpers.new_choice_modal_catalog_admin_choice_set_path(catalog, I18n.locale, choice_set),
         createChoiceUrl: Rails.application.routes.url_helpers.catalog_admin_choice_set_choices_path(catalog, I18n.locale, choice_set),
-        fetchUrl: Rails.application.routes.url_helpers.react_choices_for_choice_set_path(
-          catalog.slug,
-          I18n.locale,
-          item_type.slug,
-          field_uuid: uuid,
-          choice_set_id: choice_set.id
-        ),
+        fetchUrl: if item_type.is_a?(Category)
+                    Rails.application.routes.url_helpers.react_category_choices_for_choice_set_path(
+                      catalog.slug,
+                      I18n.locale,
+                      item_type.id,
+                      field_uuid: uuid,
+                      choice_set_id: choice_set.id
+                    )
+                  else
+                    Rails.application.routes.url_helpers.react_choices_for_choice_set_path(
+                      catalog.slug,
+                      I18n.locale,
+                      item_type.slug,
+                      field_uuid: uuid,
+                      choice_set_id: choice_set.id
+                    )
+                  end,
```

Comment:
This is inspired from : https://github.com/catima/catima/blob/51df79068aff3f41eb014708f82fe2bd86e45488/app/models/field/choice_set.rb#L233-L253